app/Main.hs

```diff
@@ -67,13 +75,22 @@ readSource = \case
   FileInput path -> TIO.readFile path
   ExecInput str -> pure (T.pack str)
 
+mkExtVar :: (Text, Either ExtVarType ExtVar) -> IO (Text, ExtVar)
```

Comment:
I am wondering could we put this into a record, where we have the var name, type, and a `Maybe` value? That way the type would be a bit easier to understand.

Reply:
I was thinking about a potential refactor on this, I am surprised to see the PR got merged :smile: Anyway, we have a little explanation about the `Either ExtVarType ExtVar` type at the definition of `Options`, so I don't think it will be too much of a problem as it is.

Reply:
Hey no problem, you can do some refactoring but I think it's not needed honestly. I only would like types to be a bit more descriptive. Better yet, would you like to pick some other ticket? 🤭